src/main/java/org/wildfly/security/sasl/digest/DigestClientFactory.java

```diff
@@ -60,6 +62,9 @@ public DigestClientFactory(final Provider provider) {
     public SaslClient createSaslClient(String[] mechanisms, String authorizationId, String protocol, String serverName,
             Map<String, ?> props, CallbackHandler cbh) throws SaslException {
 
+        if (cbh == null) throw ElytronMessages.log.mechCallbackHandlerMissing().toSaslException();
```

Comment:
I don't think this is a realistic requirement; RuntimeExceptions are unavoidable and are preferred in the event of a programming error condition (like a null parameter).  This allows programming errors to take an exceptional path on the client or server.